core/src/unprocessed_packet_batches.rs

```diff
@@ -368,9 +374,40 @@ pub fn packet_message(packet: &Packet) -> Result<&[u8], DeserializedPacketError>
         .ok_or(DeserializedPacketError::SignatureOverflowed(sig_size))
 }
 
-/// Computes `(addition_fee + base_fee / requested_cu)` for `deserialized_packet`
-fn compute_fee_per_cu(_message: &VersionedMessage, _bank: &Bank) -> u64 {
-    1
+/// Computes `(addition_fee / requested_cu)` if feature tx_wide_compute_cap is enabled,
+/// otherwise return default 0u64
+fn compute_fee_per_cu(message: &VersionedMessage, bank: &Bank) -> u64 {
+    if bank.feature_set.is_active(&tx_wide_compute_cap::id()) {
+        if let Some(sanitized_message) = sanitize_message(message, bank) {
+            let mut compute_budget = ComputeBudget::default();
+            match compute_budget.process_message(&sanitized_message, false, true) {
+                Ok(requested_additional_fee) => requested_additional_fee / compute_budget.max_units,
+                _ => 0u64,
+            };
+        }
+    }
+
+    0u64
+}
+
+fn sanitize_message(
+    versioned_message: &VersionedMessage,
+    address_loader: impl AddressLoader,
+) -> Option<SanitizedMessage> {
+    versioned_message.sanitize().ok()?;
+
+    match versioned_message {
+        VersionedMessage::Legacy(message) => Some(SanitizedMessage::Legacy(message.clone())),
+        VersionedMessage::V0(message) => {
+            let loaded_addresses = address_loader
+                .load_addresses(&message.address_table_lookups)
+                .ok()?;
+            Some(SanitizedMessage::V0(v0::LoadedMessage::new(
+                message.clone(),
+                loaded_addresses,
+            )))
+        }
+    }
```

Comment:
If we're going to sanitize every transaction here, can we avoid sanitizing when we process in `process_packets_transactions` https://github.com/carllin/solana/blob/1dc2780289c21bd1efce43ccb49021a49a086c44/core/src/banking_stage.rs#L1830-L1835

Reply:
nvm, it seems like these `SanitizedMessages` are specific to the context of a specific bank/slot: https://github.com/jeffwashington/solana/blob/a44f3c4dbeb5d8740d2029df5e26f71852be0a9c/runtime/src/accounts.rs#L574-L578, i.e. loading from a different slot could yield a different account